npm/test/basics.js

```diff
@@ -879,3 +880,251 @@ test("debug service compiling multiple sources - web worker", async () => {
     debugService.terminate();
   }
 });
+
+test("CreateIntegerTicks: invalid inputs", () => {
+  runAndAssertIntegerTicks(2, 1, []);
+  runAndAssertIntegerTicks(0, 2, []);
+  runAndAssertIntegerTicks(-5, 100, []);
+});
+
+test("CreateIntegerTicks: below 100", () => {
+  runAndAssertIntegerTicks(1, 1, [1]);
+  runAndAssertIntegerTicks(3, 3, [3]);
+  runAndAssertIntegerTicks(4, 6, [4, 5, 6]);
+  runAndAssertIntegerTicks(1, 100, [1, 10, 100]);
+  runAndAssertIntegerTicks(1, 10, [1, 10]);
+  runAndAssertIntegerTicks(1, 9, [1]);
+  runAndAssertIntegerTicks(2, 10, [10]);
+  runAndAssertIntegerTicks(2, 9, [2, 3, 4, 5, 6, 7, 8, 9]);
+});
+
+test("CreateIntegerTicks: more than 100", () => {
+  runAndAssertIntegerTicks(20, 59, [20, 30, 40, 50]);
+  runAndAssertIntegerTicks(231, 365, [300]);
+  runAndAssertIntegerTicks(331, 365, [340, 350, 360]);
+  runAndAssertIntegerTicks(567, 569, [567, 568, 569]);
+});
+
+test("CreateIntegerTicks: expected qubit numbers", () => {
+  runAndAssertIntegerTicks(400, 8000000, [1000, 10000, 100000, 1000000]);
+  runAndAssertIntegerTicks(12345, 67890, [20000, 30000, 40000, 50000, 60000]);
+  runAndAssertIntegerTicks(23456, 27890, [24000, 25000, 26000, 27000]);
+});
+
+test("CreateTimeTicks: invalid inputs", () => {
+  runAndAssertTimeTicks(2, 1, []);
+  runAndAssertTimeTicks(0, 2, []);
+  runAndAssertTimeTicks(-5, 100, []);
+});
+
+test("CreateTimeTicks: nanoseconds below 100", () => {
+  runAndAssertTimeTicks(1, 1, ["nanosecond"]);
+  runAndAssertTimeTicks(3, 3, ["3 nanoseconds"]);
+  runAndAssertTimeTicks(4, 6, [
+    "4 nanoseconds",
+    "5 nanoseconds",
+    "6 nanoseconds",
+  ]);
+  runAndAssertTimeTicks(1, 100, ["nanosecond"]);
```

Comment:
Should we prefix the singular with the value as we do for the plural (e.g. "1 nanosecond")?

Reply:
I thought to follow https://quantum.microsoft.com/en-us/experience/quantum-cryptography where we use units without values. But I do not have a strong preference.